### PR TITLE
Highlight @You mentions

### PR DESCRIPTION
**Pull request by ceokot**

### CODE REVIEW

**Comment by KeeJef** [comment]
Was this completed in theming on Android?

**Comment by ceokot** [comment]
> Was this completed in theming on Android?

This wasn't completed in theming and here neither before we shifted to disappearing messages rebuild.

**Comment by KeeJef** [comment]
Can this be taken out of draft @0x330a ? Any merge conflicts, if so might be a good candidate for the point release

**Comment by mpretty-cyro** [comment]
This will be folded into another branch which involves cleaning up some of the MentionUtilities logic

**Comment by mpretty-cyro** [comment]
Reopening this until it's decided which PR it's going to be combined into

File: app/src/main/java/org/thoughtcrime/securesms/util/RoundedBackgroundSpan.kt [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+ * @param cornerRadius - The corner radius of the background in pixels. Defaults to 8dp.
+ * @param paddingHorizontal - The horizontal padding of the text in pixels. Defaults to 3dp.
+ * @param paddingVertical - The vertical padding of the text in pixels. Defaults to 3dp.
+ */
+
+
+class RoundedBackgroundSpan(
+    private val textColor: Int,
+    private val backgroundColor: Int,
+    private val cornerRadius: Float = toPx(8, Resources.getSystem()).toFloat(), // setting some Session defaults
```
Comment (simophin) [inline app/src/main/java/org/thoughtcrime/securesms/util/RoundedBackgroundSpan.kt:23]: This is the first time I see the usage of `Resources.getSystem()`, is it safe to use, in terms of UI scale changing, themeing POV? From the doco I see this:

> Return a global shared Resources object that provides access to only system resources (no application resources), is not configured for the current screen (can not use dimension units, does not change based on orientation, etc), and is not affected by Runtime Resource Overlay.

File: app/src/main/java/org/thoughtcrime/securesms/util/RoundedBackgroundSpan.kt [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        canvas: Canvas, text: CharSequence, start: Int, end: Int,
+        x: Float, top: Int, y: Int, bottom: Int, paint: Paint
+    ) {
+        // the top needs to take into account the font and the required vertical padding
+        val newTop = y + paint.fontMetrics.ascent - paddingVertical
+        val newBottom = y + paint.fontMetrics.descent + paddingVertical
+        val rect = RectF(
+            x,
+            newTop,
+            x + measureText(paint, text, start, end) + 2 * paddingHorizontal,
```
Comment (simophin) [inline app/src/main/java/org/thoughtcrime/securesms/util/RoundedBackgroundSpan.kt:38]: Would it be expensive to do measuring while drawing? I thought the measure should already be done at this point and the boundary passed in the `draw` method should be sufficient?

File: app/src/main/java/org/thoughtcrime/securesms/util/RoundedBackgroundSpan.kt (hunk 1 above)
Comment (ThomasSession) [reply app/src/main/java/org/thoughtcrime/securesms/util/RoundedBackgroundSpan.kt:23]: Yeah I was a bit surprised by this as well. That line was there from the original PR and I moved it around, but not sure how good this is. I did test it across multiple themes and it seems fine, but I'm also happy to change it.

Comment (ThomasSession) [reply app/src/main/java/org/thoughtcrime/securesms/util/RoundedBackgroundSpan.kt:23]: It seems to be used a fair throughout the app in the existing code.

File: app/src/main/java/org/thoughtcrime/securesms/util/RoundedBackgroundSpan.kt (hunk 2 above)
Comment (ThomasSession) [reply app/src/main/java/org/thoughtcrime/securesms/util/RoundedBackgroundSpan.kt:38]: The `draw` unfortunately does not provide any information on the width beyond giving you the charsequence, which is why it needs to be calculated there.

Comment (simophin) [reply app/src/main/java/org/thoughtcrime/securesms/util/RoundedBackgroundSpan.kt:38]: My bad I didn't see that the `draw` doesn't give you this info.

**Review by simophin — APPROVED** [review]
LGTM

File: app/src/main/java/org/thoughtcrime/securesms/util/RoundedBackgroundSpan.kt (hunk 1 above)
Comment (ThomasSession) [reply app/src/main/java/org/thoughtcrime/securesms/util/RoundedBackgroundSpan.kt]: Removed it.

**Pull request merged**